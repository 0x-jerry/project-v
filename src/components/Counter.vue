<script lang="ts" setup>
defineProps<{
  modelValue: number
}>()

defineEmits<{
  (type: 'update:modelValue', val: number): void
}>()
</script>

<template>
  <div>
    count: {{ modelValue }}
    <button @click="$emit('update:modelValue', modelValue + 1)">+</button>
    <button @click="$emit('update:modelValue', modelValue - 1)">-</button>
  </div>
</template>
