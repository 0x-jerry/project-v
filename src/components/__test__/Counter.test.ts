import { mount } from '@vue/test-utils'
import Counter from '../Counter.vue'

test('mount component', async () => {
  expect(Counter).toBeTruthy()

  const wrapper = mount(Counter, {
    props: {
      modelValue: 4,
    },
  })

  expect(wrapper.find('span').text()).toContain('4')
  expect(wrapper.html()).toMatchSnapshot()
})
