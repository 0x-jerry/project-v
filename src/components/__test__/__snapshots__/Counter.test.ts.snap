// Vitest Snapshot v1

exports[`mount component 1`] = `"<div><span> count: 4</span><button>+</button><button>-</button></div>"`;
